### DISCUSSION TITLE
DSK persistent stat on status

### DISCUSSION THREAD

**Discussion by milotalpo** [ericaltendorf/plotman#468] [Q&A]
I'm new to plotman and Chia (and github btw) and I am currently trying to plot 12 parallel plots with plotman on a machine running Ubuntu Server.
Checking `plotman interactive` and/or `plotman status`, every plot after phase 2 gives a fairly persistent `DSK` stat - i.e. every 1-2 minutes the stat turns to `RUN` for a few seconds and after that it goes back to `DSK`.

Quoting plotman Wiki at DSK:

> The disk itself is asleep and not running.

I doubled checked and 1.the disks are effectively accessible 2.USB selective autosuspend is disabled 3.the machine is normally plotting, as I already have 3 plots done _but_ in a VERY loooong time (12 hours for the first plot, last one 17 hours).

I'm wondering if that is an issue with _too many parallel plots at the same time_ and/or a misconfiguration in plotman.yaml.

Current machine configuration:
1xRyzen 7 3700x, 64gb DDR4 RAM, 2xNVME 2tb (consumer grade) with RAID-0 and 4x8tb Seagate Backup USB 3.0 HDDs.

Current plotman.yaml config:
```
tmpdir_stagger_phase_major: 2
tmpdir_stagger_phase_minor: 1
tmpdir_stagger_phase_limit: 4
tmpdir_max_jobs: 8
global_max_jobs :12
global_stagger_m: 45
polling_time_s: 10
plotting:
k: 32
e: false
n_threads: 4
n_buckets: 128
job_buffer: 4000
```
A big thanks to anyone willing to help!

![abcd](https://user-images.githubusercontent.com/84225399/118354828-45ad9a80-b56d-11eb-8d3d-e2ed8e6dd63a.JPG)

**Comment by BrunoCampana** [comment]
Hello @milotalpo , did you manage to solve this issue of plots at "DSK" stats? I'm having the exactly same issue, also in Ubuntu, and my daily plot/rate is being quite lower than what my HW can manage. I think this "DSK issue", i.e., SSD NVMe being suspend might be the issue.

**Reply by milotalpo** [reply]
Hi. Nope, tried everything (reinstalling plotman, reconfiguring it, using 1 nvme instead of 2 raid 0'ed..) but nothing changed and since the only thing left to try was to change the hardware I gave up, sold everything and bought some crypto with the money 😂

**Comment by thevidocq** [comment]
Any new? I have the same problem.

Thanks!

**Comment by altendky** [comment]
That description of `DSK` seems completely wrong.  I'll have to review that.  For hangs in the Chia Network plotter I would be going to keybase for help.  Sorry for the extremely delayed response here.

**Reply by altendky** [reply]
But maybe I misread and this isn't about hangs...  My plots take awhile too.  When you make your computer do a lot of work each task takes longer than when you only do one.  Often, a lot longer.  Without knowing the specific disk at least it'll be hard.  But, it's probably best to check in `#plotting` on Chia's keybase for some interactive help.

**Reply by altendky** [reply]
For example, the first (not even done all alone) plot when I last ramped up one of my plotters was 8.5 hours or so.  Normal runtime fully loaded is over 14 hours.

**Reply by graemes** [reply]
Looking at the config I'd say it is an issue with the number of plots running in parallel.  From the screenshot - there are 20 processing threads running in P1 + another 7 in later phases.  This is on a CPU with 8 physical cores.  This works out to be approx. 3.5 heavy processing threads per core.  Think the poor machine is thrashing causing the DSK message (which I would guess is actually saying it is waiting for data from the disk).

**Reply by graemes** [reply]
What would be interesting is seeing what iostat displayed or the iowaits in top.

**Reply by altendky** [reply]
Isn't `iowait` the last column?  Showing up to 10 hours.  So yeah, way overloaded if that's an 8-core CPU and maybe also way overloading the NVMEs depending what they are.  On the system I mentioned above I get up over 1 hour.  On my other plotter with 4x 1tb Inland with RAID0 it gets to maybe 15 seconds with 21 jobs going.
